pkg/testutil/prometheus.go

```diff
@@ -0,0 +1,97 @@
+package testutil
+
+import (
+	"fmt"
+	"io/ioutil"
+	"math"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"syscall"
+	"time"
+
+	"github.com/prometheus/tsdb"
+)
+
+// Prometheus represents a test instance for integration testing.
+// It can be populated with data before being started.
+type Prometheus struct {
+	dir     string
+	addr    string
+	running bool
+	db      *tsdb.DB
+	cmd     *exec.Cmd
+}
+
+// NewPrometheus creates a new test Prometheus instance that will listen on address.
+func NewPrometheus(address string) (*Prometheus, error) {
+	dir, err := ioutil.TempDir("", "prometheus-test")
+	if err != nil {
+		return nil, err
+	}
+	// Just touch an empty config file. We don't need to actually scrape anything.
+	_, err = os.Create(filepath.Join(dir, "prometheus.yml"))
+	if err != nil {
+		return nil, err
+	}
+
+	db, err := tsdb.Open(dir, nil, nil, &tsdb.Options{
+		WALFlushInterval:  10 * time.Millisecond,
+		BlockRanges:       []int64{2 * 3600 * 1000},
+		RetentionDuration: math.MaxInt64,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	return &Prometheus{
+		dir:  dir,
+		addr: address,
+		db:   db,
+	}, nil
+}
+
+// Start runs the Prometheus instance until the context is canceled.
```

Comment:
wait.. where is that context? ;p

Reply:
I think it was before... so now it is just `Stop()`

Reply:
Yes!